galsim/chromatic.py

```diff
@@ -792,38 +819,30 @@ class InterpolatedChromaticObject(ChromaticObject):
                             whichever wavelength has the highest Nyquist frequency.
                             `oversample_fac`>1 results in higher accuracy but costlier
                             pre-computations (more memory and time). [default: 1]
+    @param use_exact_SED    If true, then rescale the interpolated image for a given wavelength by
+                            the ratio of the exact SED at that wavelength to the linearly
+                            interpolated SED at that wavelength.  Thus, the flux of the interpolated
+                            object should be correct, at the possible expense of other features.
+                            [default: True]
     """
-    def __init__(self, original, waves, oversample_fac=1.0):
+    def __init__(self, original, waves, oversample_fac=1.0, use_exact_SED=True):
+
+        self.waves = np.sort(np.array(waves))
+        self.oversample = oversample_fac
+        self.use_exact_SED = use_exact_SED
 
         self.separable = original.separable
-        if self.separable:
-            self.SED = original.SED
+        self.interpolated = True
+        self.SED = original.SED
         self.wave_list = original.wave_list
 
         # Don't interpolate an interpolation.  Go back to the original.
-        if isinstance(original, InterpolatedChromaticObject):
-            original = original.original
-        self.original = original
-        self.waves = np.sort(np.array(waves))
-        self.oversample = oversample_fac
+        self.original = original.deinterpolated
```

Comment:
>Why go through this rigamarole with the deinterpolated property

At one point I had had some thoughts about making a default `ChromaticObject.deinterpolated` property so you wouldn't need to define it in every subclass, but now I think you're right, it's easier to just use an attribute.  Done.